comparison/ansible/roles/grub_vswitch/tasks/main.yml

```diff
@@ -1,9 +1,17 @@
 ---
-- name: Grub Configuration for vSwitch
+- name: Grub Configuration for vSwitch (Default)
   lineinfile:
     dest: /etc/default/grub
     regexp: ^GRUB_CMDLINE_LINUX=
     line: GRUB_CMDLINE_LINUX="console=tty0 console=ttyS1,115200n8 biosdevname=0 net.ifnames=1 numa_balancing=disable intel_pstate=disable intel_iommu=on iommu=pt isolcpus=2-27,30-55 nohz_full=2-27,30-55 rcu_nocbs=2-27,30-55 nmi_watchdog=0 audit=0 nosoftlockup processor.max_cstate=1 intel_idle.max_cstate=1 hpet=disable tsc=reliable mce=off numa_balancing=disable hugepagesz=2M hugepages={{ hugepages }}"
+  when: not ansible_processor_cores == 16
+
+- name: Grub Configuration for vSwitch (Gold 5218)
+  lineinfile:
+    dest: /etc/default/grub
+    regexp: ^GRUB_CMDLINE_LINUX=
+    line: GRUB_CMDLINE_LINUX="console=tty0 console=ttyS1,115200n8 biosdevname=0 net.ifnames=1 numa_balancing=disable intel_pstate=disable intel_iommu=on iommu=pt isolcpus=2-27,34-59 nohz_full=2-27,34-59 rcu_nocbs=2-27,34-59 nmi_watchdog=0 audit=0 nosoftlockup processor.max_cstate=1 intel_idle.max_cstate=1 hpet=disable tsc=reliable mce=off numa_balancing=disable hugepagesz=2M hugepages={{ hugepages }}"
```

Comment:
2M hugepages are good, but 1G is better. Granularity is worse though.

Reply:
IIRC, 2M was initially used due to a bug or limitation in VPP. Switching to 1G would be better, but even with our DPDK based applications I wouldn't expect to see any difference in observable performance.

If we are to switch to 1G pages as default, this should be done through a separate issue/PR, as it will likely require updates to all of our VPP based infra (vSwitch) and use-cases (CNFs)

Reply:
It would be nice to have the grub command line templated out and the values were generated dynamically including the CPU assignments.

Reply:
Yep, that would be a good addition to the existing (very static) role. I know Peter M (csit) did some work in this area with the tools/cpu_util.sh script. Something similar might be useful for preparing hosts with CPU isolation.